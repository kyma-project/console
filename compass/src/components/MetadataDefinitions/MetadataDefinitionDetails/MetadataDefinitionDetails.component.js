import React, { useState } from 'react';
import { ActionBar } from 'fundamental-react';
import {
  Button,
  Breadcrumb,
  Panel,
  PanelHead,
  PanelHeader,
  PanelBody,
} from '@kyma-project/react-components';
import LuigiClient from '@kyma-project/luigi-client';

import '../../../shared/styles/header.scss';
import ResourceNotFound from '../../Shared/ResourceNotFound.component';
import JSONEditorComponent from '../../Shared/JSONEditor';

const MetadataDefinitionDetails = ({
  metadataDefinition: metadataDefinitionQuery,
  updateLabelDefinition,
}) => {
  const [isSchemaValid, setSchemaValid] = useState(true);
  const [schema, setSchema] = useState({});

  const handleSchemaChange = schema => {
    try {
      setSchema(JSON.parse(schema));

      setSchemaValid(true);
    } catch {
      setSchemaValid(false);
    }
  };

  const handleSaveChanges = async definitionKey => {
    try {
      await updateLabelDefinition({
        key: definitionKey,
        schema,
      });
      metadataDefinitionQuery.refetch(); //  to format the JSON
      // onCompleted(runtimeName, `Runtime created succesfully`);
    } catch (e) {
      //onError(`The runtime could not be created succesfully`, e.message || ``);
    }
  };

  const metadataDefinition =
    (metadataDefinitionQuery && metadataDefinitionQuery.labelDefinition) || {};
  const loading = metadataDefinitionQuery.loading;
  const error = metadataDefinitionQuery.error;

  if (!metadataDefinitionQuery) {
    if (loading) return 'Loading...';
    if (error)
      return (
        <ResourceNotFound
          resource="Metadata Definition"
          breadcrumb="MetadataDefinitions"
        />
      );
    return null;
  }
  if (error) {
    return `Error! ${error.message}`;
  }

  return (
    <>
      <header className="fd-has-background-color-background-2">
        <section className="fd-has-padding-regular fd-has-padding-bottom-none action-bar-wrapper">
          <section>
            <Breadcrumb>
              <Breadcrumb.Item
                name="Metadata Definitions"
                url="#"
                onClick={() =>
                  LuigiClient.linkManager()
                    .fromClosestContext()
                    .navigate(`/metadata-definitions`)
                }
              />
              <Breadcrumb.Item />
            </Breadcrumb>
            <ActionBar.Header
              title={metadataDefinition.key || 'Loading name...'}
            />
          </section>
          <ActionBar.Actions>
            <Button
              onClick={() => handleSaveChanges(metadataDefinition.key)}
              disabled={!isSchemaValid}
              option="emphasized"
            >
              Save
            </Button>
          </ActionBar.Actions>
        </section>
      </header>
      <section className="fd-section">
        <Panel>
          <PanelHeader>
            <PanelHead title="Schema" />
          </PanelHeader>

          <PanelBody>
            <JSONEditorComponent
              onChangeText={handleSchemaChange}
              text={JSON.stringify(metadataDefinition.schema || {}, null, 2)}
            />
          </PanelBody>
        </Panel>
      </section>
    </>
  );
};

export default MetadataDefinitionDetails;
