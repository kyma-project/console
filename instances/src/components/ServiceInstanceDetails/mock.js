export const Mock = {
  __typename: 'clusterDocsTopic',
  name: 'kyma',
  groupName: 'root',
  displayName: 'Kyma',
  description: 'Overall documentation for Kyma',
  assets: [
    {
      __typename: 'clusterAsset',
      name: 'asset1',
      type: 'markdown',
      files: [
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/01-01-in-a-nutshell.md',
          metadata: {
            __typename: 'fileMetadata',
            title: 'In a nutshell',
            type: 'Overview',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/04-01-overview.md',
          metadata: {
            __typename: 'fileMetadata',
            title: 'Overview',
            type: 'Installation',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/04-03-cluster-installation.md',
          metadata: {
            __typename: 'fileMetadata',
            // title: 'Install Kyma on a cluster',
            type: 'Installation',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/12-01-try-out-kyma.md',
          metadata: {
            __typename: 'fileMetadata',
            type: 'Examples',
            title: 'Try out Kyma',
          },
        },
      ],
    },
  ],
};
