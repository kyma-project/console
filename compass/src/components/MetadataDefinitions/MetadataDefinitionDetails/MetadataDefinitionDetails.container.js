import { graphql, compose } from 'react-apollo';

import {
  DELETE_LABEL_DEFINITION,
  GET_LABEL_DEFINITION,
  UPDATE_LABEL_DEFINITION,
} from '../gql';
import { SEND_NOTIFICATION } from '../../../gql';

import MetadataDefinitionDetails from './MetadataDefinitionDetails.component';

export default compose(
  graphql(GET_LABEL_DEFINITION, {
    name: 'metadataDefinition',
    options: props => {
      return {
        fetchPolicy: 'cache-and-network',
        errorPolicy: 'all',
        variables: {
          key: props.definitionKey,
        },
      };
    },
  }),
  graphql(UPDATE_LABEL_DEFINITION, {
    props: ({ mutate, error }) => ({
      updateLabelDefinition: data =>
        mutate({
          variables: {
            in: data,
          },
        }),
    }),
  }),
  graphql(DELETE_LABEL_DEFINITION, {
    props: ({ mutate }) => ({
      deleteLabelDefinition: key =>
        mutate({
          variables: {
            key: key,
          },
        }),
    }),
  }),
  graphql(SEND_NOTIFICATION, {
    name: 'sendNotification',
  }),
)(MetadataDefinitionDetails);
