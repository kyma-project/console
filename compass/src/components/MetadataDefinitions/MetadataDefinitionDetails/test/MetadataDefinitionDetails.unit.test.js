import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/test-utils';

import { mocks } from './mock';

import MetadataDefinitionDetails from '../MetadataDefinitionDetails.container';

const wait = require('waait');

describe('MetadataDefinitionDetails', () => {
  it('Renders null schema', async () => {
    console.warn = jest.fn();
    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <MetadataDefinitionDetails definitionKey="noschemalabel" />
      </MockedProvider>,
    );
    await wait(0); // wait for response
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // catch "Warning: componentWillReceiveProps has been renamed
    expect(console.warn.mock.calls.length).toBe(1);
    expect(console.warn.mock.calls[0][0]).toMatchSnapshot();
  });
});
