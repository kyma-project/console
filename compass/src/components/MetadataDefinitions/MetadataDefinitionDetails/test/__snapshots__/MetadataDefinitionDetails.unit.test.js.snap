// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetadataDefinitionDetails Renders null schema 1`] = `
Array [
  <header
    className="fd-has-background-color-background-2"
  >
    <section
      className="fd-has-padding-regular fd-has-padding-bottom-none action-bar-wrapper"
    >
      <section>
        <ul
          className="fd-breadcrumb"
        >
          <li
            className="fd-breadcrumb__item"
            onClick={[Function]}
          >
            <a
              className="fd-breadcrumb__link"
              href="#"
            >
              Metadata Definitions
            </a>
          </li>
          <li
            className="fd-breadcrumb__item"
          />
        </ul>
        <div
          className="fd-action-bar__header"
        >
          <h3
            className="fd-action-bar__title"
          >
            noschemalabel
          </h3>
        </div>
      </section>
      <div
        className="fd-action-bar__actions"
      >
        <button
          className="fd-button--emphasized"
          data-test-id="save"
          disabled={false}
          onClick={[Function]}
        >
          Save
        </button>
        <button
          className="fd-button--light fd-button--negative"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </section>
  </header>,
  <section
    className="fd-section"
  >
    <div
      className="fd-panel"
    >
      <div
        className="fd-panel__header"
      >
        <div
          className="fd-panel__head"
        >
          <h1
            className="fd-panel__title"
          >
            Schema
          </h1>
        </div>
        <div
          className="fd-panel__actions"
        >
          <div
            className="fd-form-item"
          >
            <label
              className="fd-form-label"
            >
              <span
                className="fd-toggle"
              >
                <input
                  checked={false}
                  className="fd-toggle__input"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="fd-toggle__switch"
                  role="presentation"
                />
              </span>
              
            </label>
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;

exports[`MetadataDefinitionDetails Renders null schema 2`] = `
"Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run \`npx react-codemod rename-unsafe-lifecycles\` in your project source folder.

Please update the following components: Mutation, Query"
`;
